frontend/src/OneBuilding/index.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+  Text, View, SafeAreaView, StyleSheet,
+  Modal, ActivityIndicator,
+} from 'react-native';
+import PropTypes from 'prop-types';
+
+import ModalHeader from './ModalHeader';
+
+import { fetchOneBuilding } from './actionCreator';
+import BuildingItem from './BuildingItem';
+import colors from '../common/colors';
+
+import TownhallDetail from './TownhallDetail';
+import AcademyDetail from './AcademyDetail';
+import FarmDetail from './FarmDetail';
+import MineDetail from './MineDetail';
+
+const styles = StyleSheet.create({
+  background: {
+    flex: 1,
+    backgroundColor: 'rgba(0,0,0,0.6)',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  container: {
+    width: 300,
+    backgroundColor: 'rgb(202,202,202)',
+    borderRadius: 6,
+  },
+  mainBody: {
+    padding: 15,
+    paddingTop: 0,
+  },
+  upgradeButtonGroup: {
+    flexDirection: 'column',
+  },
+  detailInfo: {
+    flex: 1,
+    flexDirection: 'row',
+  },
+  iconStyle: {
+    width: 16,
+    height: 16,
+  },
```

Comment:
AcademyButtons, AcademyDetails, FarmDetails, MineDetails, TownhallDetails, 
(src/Troops/Troopinformation too but you don't have to deal with that)

please reduce redundant code (use common styles) 
and check other stylings as well to reduce them